homeassistant/components/thermostat/homematic.py

```diff
@@ -0,0 +1,142 @@
+CONF_DEVICES = 'devices'
+CONF_ID = 'id'
+CHANNEL = 4
+PROPERTY_SET_TEMPERATURE = 'SET_TEMPERATURE'
+PROPERTY_VALVE_STATE = 'VALVE_STATE'
+PROPERTY_ACTUAL_TEMPERATURE = 'ACTUAL_TEMPERATURE'
+PROPERTY_BATTERY_STATE = 'BATTERY_STATE'
+PROPERTY_CONTROL_MODE = 'CONTROL_MODE'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Homematic thermostat. """
+
+    devices = []
+    try:
+        homegear = ServerProxy(config[CONF_ADDRESS])
+        for name, device_cfg in config[CONF_DEVICES].items():
+            # get device description to detect the type
+            device_type = homegear.getDeviceDescription(
+                device_cfg[CONF_ID] + ':-1')['TYPE']
+
+            if device_type in ['HM-CC-RT-DN', 'HM-CC-RT-DN-BoM']:
+                devices.append(HomematicThermostat(homegear,
+                                                   device_cfg[CONF_ID],
+                                                   name, 4))
+            elif device_type == 'HM-TC-IT-WM-W-EU':
+                devices.append(HomematicThermostat(homegear,
+                                                   device_cfg[CONF_ID],
+                                                   name, 2))
+            else:
+                raise ValueError(
+                    "Device Type '{}' currently not supported".format(
+                        device_type))
+    except socket.error:
+        _LOGGER.exception("Connection error to homematic web service")
+        return False
+
+    add_devices(devices)
+
+    return True
+
+
+class HomematicThermostat(ThermostatDevice):
+    """ Represents a Homematic thermostat. """
+
+    def __init__(self, device, _id, name, channel):
+        self.device = device
+        self._id = _id
+        self._channel = channel
+        self._name = name
+        self._full_device_name = '{}:{}'.format(self._id, self._channel)
+
+        self._current_temperature = None
+        self._target_temperature = None
+        self.update()
+
+    @property
+    def name(self):
+        """ Returns the name of the Homematic device. """
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement this thermostat expresses itself in. """
+        return TEMP_CELCIUS
+
+    @property
+    def current_temperature(self):
+        """ Returns the current temperature. """
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """ Returns the temperature we try to reach. """
+        return self._target_temperature
+
+    def set_temperature(self, temperature):
+        """ Set new target temperature """
+        self.device.setValue(self._full_device_name,
+                             PROPERTY_SET_TEMPERATURE,
+                             temperature)
+
+    @property
+    def device_state_attributes(self):
+        valve = self.device.getValue(self._full_device_name,
```

Comment:
All fetching of data should happen in the `update` method. So please cache these values along with `actual_temperature` and `target_temperature`.